src/content/blog/node-test-migration.mdx

```diff
@@ -0,0 +1,129 @@
+
+Once the PR was ready, we noticed that _Node.js test runner_ **was way slower** than Mocha. We investigated, and we discovered that Node.js spawns a new process for each test file to assure that each testing suite is run in **isolation**. Running a testing suite in isolation is, generally, a good practice, because it assures that tests run in an unpolluted environment.
+
+However, our testing suites are already isolated, in fact we were able to run our testing suites using the main thread with Mocha, without running into issues: side effects, polluted environments, etc. Unfortunately, Node.js didn't provide an option to _run all tests in the same thread_, so we have to come up with a solution (aren't we engineers, after all? we solve problems!).
+
+Using our internal `astro-scripts test` command, we are able to workaround this by creating a _temporary file that **imports** all the testing suites_, and we let Node.js test that single file. This way, only one process is spawned for the file, and we reach the same level of performance as if we were using the main process.
+
+However, this **came with a downside**: if there's a test failure or a timeout, we aren't able to tell which test is the cause. This was the main quirk we found, and we accepted the trade-off. After all, we also accepted Mocha's trade-offs!
+
+### Node.js `assert` and `chai`
+
+During the migration, we had to remove the `chai` library for `node:assert/strict`. This task uncovered that with `chai`, you can execute the same check in different ways. For example, you can run an equality check at least in **four** different ways:
+
+```js
+import { expect } from "chai";
+
+expect("foo").to.eq("foo")
+expect("foo").to.be.eq("foo")
+expect("foo").to.equal("foo")
+expect("foo").to.be.equal("foo")
+```
+
+From one point of view, it's good to have this kind of flexibility, but on the other hand the code of the tests becomes inconsistent. With the Node.js assertion module, you do this kind of check only in one way:
+
+```js
+import { assert } from "node:assert/strict";
+
+assert.equal("foo", "foo")
+```
+
+The Node.js assertion module **provides almost all functionalities** we required, so the migration from `chai` wasn't as painful as we thought. Our usage of `chai` was very minimal. However, **we miss** the `.includes` shortcut of `chai`:
+
+```js
+import { expect } from "chai";
+
+expect("It's a fine day").includes("fine")
+```
+
+The Node.js assertion module doesn't provide such utility, so we ended up using the equality assertion with the `String#includes` function:
+
+```js
+import assert from "node:assert/strict";
+
+assert.equal("It's a fine day".includes("fine"), true)
+```
+
+### Here comes the dragons
+
+As mentioned before, we have a lot of test files, and we add new tests almost every day. Opening a one-off PR that does the migration of the whole monorepo is unfeasible, it would require a lot of work from one person, and keeping the branch updated can be stressful.
```

Comment:
Small cut ✂️ 
```suggestion
As mentioned before, we have a lot of test files, and we add new tests almost every day. Opening a one-off PR that migrates the whole monorepo is unfeasible, it would require a lot of work from one person, and keeping the branch updated can be stressful.
```